server/api/crud/workflows.py

```diff
@@ -233,6 +240,11 @@ def run(
             run_name=runner.metadata.name,
             load_only=load_only,
         )
+        # We want to store the secret params as k8s secret, so later we can access them with the project internal secret
+        # key that was created.
+        server.api.api.utils.mask_notification_params_on_task_object(
+            run_spec, server.api.constants.MaskOperations.CONCEAL
+        )
```

Comment:
there is some common code here together with node selector enrichment with the schedule flow can be merged to method